src/test/mgr/test_progress.py

```diff
@@ -0,0 +1,212 @@
+#python unit test
+import unittest
+import os
+import sys
+from mock import Mock
+import pytest
+import json
+os.environ['UNITTEST'] = "1"
+sys.path.insert(0, "../../pybind/mgr")
+from progress import module
+
+class TestPgRecoveryEvent(object):
+    # Testing PgRecoveryEvent class
+
+    def setup(self):
+        # Creating the class and Mocking 
+        # a bunch of attributes for testing
+        self.test_class = module.PgRecoveryEvent
+        self._pgs = [module.PgId(1,i) for i in range(3)]
```

Comment:
these are all implementation details of the PgRecoveryEvent class - you need an instance of that to test with, not changes to the TestPgRecoveryEvent class